hostname: Add support of static hostname modification

cathay4t opened the pull request:
Example yaml:

```yaml
---
hostname:
  running: "nmstate-test.example.org"
  config: "nmstate-test.example.org"
```

The `running` is the activating hostname. The `config` is the on-disk
hostname in /etc/hostname. Then setting empty string to `config`,
nmstate will remove on-disk hostname.

For query, the nispor plugin provide both running and config hostname.
For setting, nispor plugin set the running hostname and NetworkManager
set the on disk hostname via
`org.freedesktop.NetworkManager.Settings.SaveHostname`.

Example added: `static_hostname.yml`, `dynamic_hostname.yml`

Integration test case included but skipped in CI as NetworkManager
cannot change hostname insider of container with error:

        org.freedesktop.NetworkManager.Settings.Failed: Saving the hostname failed

qinqon commented:
At `hostnamectl` there is the following concept, I don't know if NetworkManager has it:

```bash
There is also the following concept on hostnamectl, don't know if we have that at NM.

     --transient         Only set transient hostname
     --static            Only set static hostname
     --pretty            Only set pretty hostname
```

What I see that is a NetworkManager is de deactivatio of receiving it over DHCP, maybe we can implemen that here ?

```
ipv4.dhcp-send-hostname:            yes
ipv4.dhcp-hostname:                 --
```

We have to denote that this is changing the static hostname as stated on NetworkManager docs https://developer-old.gnome.org/NetworkManager/stable/nmcli.html

Let's say so at the yaml
```yaml
static-hostname:
  running: "nmstate-test.example.org"
  config: "nmstate-test.example.org"
```

Or maybe we show the running one and the static and we can change the static:
```yaml
hostname:
  running: "nmstate-test.example.org"
  static: "nmstate-test.example.org"
```

cathay4t commented:
@qinqon The `running` and `config` is aligning with DNS and routes. If you want DHCP hostname, setting config as empty would works.

ffmancera reviewed (changes requested):
Please, check CI tests. They are failing on hostnames tests. Thank you!

cathay4t committed:
hostname: Add support of static hostname modification

Example yaml:

```yaml
---
hostname:
  running: "nmstate-test.example.org"
  config: "nmstate-test.example.org"
```

The `running` is the activating hostname. The `config` is the on-disk
hostname in /etc/hostname. Then setting empty string to `config`,
nmstate will remove on-disk hostname.

For query, the nispor plugin provide both running and config hostname.
For setting, nispor plugin set the running hostname and NetworkManager
set the on disk hostname via
`org.freedesktop.NetworkManager.Settings.SaveHostname`.

Example added: `static_hostname.yml`, `dynamic_hostname.yml`

Integration test case included but skipped in CI as NetworkManager
cannot change hostname insider of container with error:

    org.freedesktop.NetworkManager.Settings.Failed: Saving the hostname failed

Signed-off-by: Gris Ge <fge@redhat.com>
